### PR TITLE
feat(UX-881): Range Selector

### PR DESCRIPTION
**Pull request by DE7924**

### CODE REVIEW

File: test/src/components/range_selector/range_selector_test.dart [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+import '../../../test_utils/test_app.dart';
+import '../../../test_utils/tolerant_comparator.dart';
+import '../../../test_utils/utils.dart';
+
+void main() {
+  const String parentFolder = 'range_selector';
+
+  const goldenFile = GoldenFiles(component: parentFolder);
+  setUpAll(() {
+    goldenFileComparator = TolerantComparator(goldenFile.uri, tolerance: 0.001);
```
Comment (thelukewalton) [inline test/src/components/range_selector/range_selector_test.dart:14]: why are you using a different tolerance?

File: example/widgetbook/pages/components/range_selector_widgetbook.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  Widget build(BuildContext _) {
+    return Padding(
+      padding: const EdgeInsets.all(16),
+      child: ZetaRangeSelector(
+        label: context.knobs.string(label: 'Label', initialValue: 'Range Selector'),
+        divisions: context.knobs.intOrNull.input(label: 'Divisions', initialValue: null),
+        showValues: context.knobs.boolean(label: 'Show Values', initialValue: true),
+        onChange: disabledKnob(context) ? null : (value) {},
+        largeSliderThumbs: context.knobs.boolean(label: 'Large Slider Thumbs', initialValue: false),
+        rounded: context.knobs.booleanOrNull(label: 'Rounded', initialValue: true),
```
Comment (thelukewalton) [inline example/widgetbook/pages/components/range_selector_widgetbook.dart:28]: can you remove the rounded prop in widgetbook  - we already set rounded in the Addons panel and having this here overrides that

File: lib/src/components/range_selector/range_selector.dart [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+    super.rounded,
+    required this.initialValues,
+    this.min = 0.0,
+    this.max = 100,
+    this.label,
+    this.onChange,
+    this.divisions,
+    this.semanticLabel,
+    this.showValues = true,
+    this.largeSliderThumbs = false,
```
Comment (thelukewalton) [inline lib/src/components/range_selector/range_selector.dart:24]: I can't see the large thumbs in the designs - can we remove this until that is added there please

File: lib/src/components/range_selector/range_selector.dart [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+  final ValueChanged<RangeValues>? onChange;
+
+  /// The number of divisions for the range selector.
+  final int? divisions;
+
+  /// Value passed to the wrapping [Semantics] widget.
+  final String? semanticLabel;
+
+  /// Whether to show the values of the range selector.
+  /// Defaults to true.
```
Comment (thelukewalton) [inline lib/src/components/range_selector/range_selector.dart:54]: In general we don't _need_ to have 'Defaults to true' in the comments - the flutter analyzer knows the defaults and can pre populate this in any documentation 
image

**Review by thelukewalton — APPROVED** [review]
Wow this is a really complicated component. Well done!

File: test/src/components/range_selector/range_selector_test.dart (hunk 1 above)
Comment (thelukewalton) [reply test/src/components/range_selector/range_selector_test.dart:14]: That is interesting. We should experiment with other components to see if more can have a lower tolerance

Comment (DE7924) [reply test/src/components/range_selector/range_selector_test.dart:14]: I think it's more of a case of if certain components Should have a lower tolerance due to there design. For example if the component takes up less area in the 800x600 plane,,,, say less than 10% of the pixels (just an example) then the tolerance should be lowered as small pixel changes may not even be a 1% difference

Comment (thelukewalton) [reply test/src/components/range_selector/range_selector_test.dart:14]: I guess we should write a function to ensure the components take up the whole space in the tests

**Review by thelukewalton — APPROVED** [review]

**Pull request merged**